src/cascadia/TerminalSettingsModel/CascadiaSettings.cpp

```diff
@@ -289,25 +289,24 @@ winrt::Microsoft::Terminal::Settings::Model::Profile CascadiaSettings::Duplicate
     }
     duplicated->Name(winrt::hstring(newName));
 
-#define DUPLICATE_SETTING_MACRO(settingName)                                                                                                   \
-    if (source.Has##settingName() ||                                                                                                           \
-        (source.##settingName##OverrideSource() != nullptr && source.##settingName##OverrideSource().Origin() != OriginTag::ProfilesDefaults)) \
-    {                                                                                                                                          \
-        duplicated->##settingName(source.##settingName());                                                                                     \
-    }
+    const auto isProfilesDefaultsOrigin = [](const auto& profile) -> bool {
+        return profile && profile.Origin() != OriginTag::ProfilesDefaults;
+    };
 
-#define DUPLICATE_FONT_SETTING_MACRO(settingName)                                                                                                                                    \
-    if (source.FontInfo().Has##settingName() ||                                                                                                                                      \
-        (source.FontInfo().##settingName##OverrideSource() != nullptr && source.FontInfo().##settingName##OverrideSource().SourceProfile().Origin() != OriginTag::ProfilesDefaults)) \
-    {                                                                                                                                                                                \
-        duplicated->FontInfo().##settingName(source.FontInfo().##settingName());                                                                                                     \
+    const auto isProfilesDefaultsOriginSub = [=](const auto& sub) -> bool {
+        return sub && isProfilesDefaultsOrigin(sub.SourceProfile());
+    };
+
+#define DUPLICATE_SETTING_MACRO(settingName)                                                         \

```

Comment:
These macros where equally valid. Since I had to touch them anyways I fixed their inefficiency as a "drive by". We now use generic lambdas to abstract away as much work as possible from the macro. This doesn't just make debugging and reading easier (the macro looks a lot less scary without the diff), but also reduces code sizes after compilation, since the argument to our generic lambdas will always be the same type.